### PR TITLE
Mock window dimensions

### PR DESCRIPTION
**Pull request by Latropos**
## Summary
Mock window dimensions to make snapshot tests not rely on any parameters of actual device.

## Test plan
🥳 Tests pass on Android

image

### CODE REVIEW

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 1]
```diff
@@ -558,6 +565,46 @@ export class TestRunner {
...
+      if (global._LayoutAnimationsManager) {
+        global.LayoutAnimationsManager = global._LayoutAnimationsManager;
+      }
+    });
+  }
+
+  public async mockWindowDimensions() {
+    await this.runOnUIBlocking(() => {
+      'worklet';
+      const originalCreateAnimatedComponent = global.LayoutAnimationsManager;
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:580]:
```suggestion
      const originalLayoutAnimationsManager = global.LayoutAnimationsManager;
```

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 2]
```diff
@@ -558,6 +565,46 @@ export class TestRunner {
...
+          {
+            ..._yogaValues,
+            windowHeight: 852,
+            windowWidth: 393,
+          },
+          config,
+        );
+      };
+
+      global._LayoutAnimationsManager = originalCreateAnimatedComponent;
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:600]:
```suggestion
      global.originalLayoutAnimationsManager = originalCreateAnimatedComponent;
```

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 3]
```diff
@@ -558,6 +565,46 @@ export class TestRunner {
...
+      }
+    });
+  }
+
+  public async mockWindowDimensions() {
+    await this.runOnUIBlocking(() => {
+      'worklet';
+      const originalCreateAnimatedComponent = global.LayoutAnimationsManager;
+
+      const createAnimatedComponentOnStart: LayoutAnimationStartFunction = (
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:582]:
```suggestion
      const startLayoutAnimation: LayoutAnimationStartFunction = (
```

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 4]
```diff
@@ -558,6 +565,46 @@ export class TestRunner {
...
+            windowHeight: 852,
+            windowWidth: 393,
+          },
+          config,
+        );
+      };
+
+      global._LayoutAnimationsManager = originalCreateAnimatedComponent;
+      global.LayoutAnimationsManager = {
+        start: createAnimatedComponentOnStart,
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:602]:
```suggestion
        start: startLayoutAnimation,
```

File: packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts [hunk 5]
```diff
@@ -80,7 +80,7 @@ export type LayoutAnimationStartFunction = (
   tag: number,
   type: LayoutAnimationType,
   yogaValues: Partial<SharedTransitionAnimationsValues>,
-  config: LayoutAnimationFunction
+  config: (arg: Partial<SharedTransitionAnimationsValues>) => LayoutAnimation
```
Comment (piaskowyk) [inline packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts:83]: Why do we need to change this type? 🤔

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: packages/react-native-reanimated/src/layoutReanimation/animationsManager.ts [hunk 6]
```diff
@@ -34,7 +35,10 @@ function stopObservingProgress(
   global._notifyAboutEnd(tag, removeView);
 }
 
-function createLayoutAnimationManager() {
+function createLayoutAnimationManager(): {
+  start: LayoutAnimationStartFunction;
+  stop: (tag: number) => void;
+} {
```
Comment (Latropos) [inline packages/react-native-reanimated/src/layoutReanimation/animationsManager.ts:41]: .

File: packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts (hunk 5 above)
Comment (Latropos) [reply packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts:83]: We have defined and exported type `LayoutAnimationStartFunction`, which was expected to be used to type function `createLayoutAnimationManager`. However this type got suddenly disconnected from the function and the type wasn't used anywhere across our codebase. Here I connect them back: https://github.com/software-mansion/react-native-reanimated/pull/6235#discussion_r1678966967
For this reason the type and the function went out of sync.

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 7]
```diff
@@ -501,6 +508,46 @@ export class TestRunner {
...
+          {
+            ..._yogaValues,
+            windowHeight: 852,
+            windowWidth: 393,
+          },
+          config,
+        );
+      };
+
+      global._LayoutAnimationsManager = originalLayoutAnimationsManager;
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:543]:
```suggestion
      global.originalLayoutAnimationsManager = originalLayoutAnimationsManager;
```

**Review by piaskowyk — CHANGES_REQUESTED** [review]

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**